File: simplified-app-shared/src/main/java/org/nypl/simplified/app/ScreenSizeInformationType.java
package org.nypl.simplified.app;

/**
 * Information about the device screen.
 */

public interface ScreenSizeInformationType
{
  /**
   * @param dp A size in dp
   *
   * @return The given size converted to pixels
   */

  double screenDPToPixels(
    final int dp);

  /**
   * @return The DPI of the current screen
   */

  double screenGetDPI();

  /**
   * @return The height of the screen in pixels
   */

  int screenGetHeightPixels();

  /**
   * @return The width of the screen in pixels
   */

  int screenGetWidthPixels();

}

File: simplified-ui-screen/src/main/java/org/nypl/simplified/ui/screen/ScreenSizeInformationType.kt
package org.nypl.simplified.ui.screen

/**
 * Information about the device screen.
 */

interface ScreenSizeInformationType {

  /**
   * @param dp A size in dp
   *
   * @return The given size converted to pixels
   */

  fun dpToPixels(dp: Int): Double

  /**
   * @return `true` if the screen is currently in portrait orientation
   */

  val isPortrait: Boolean

  /**
   * @return The DPI of the current screen
   */

  val dpi: Double

  /**
   * @return The height of the screen in pixels
   */

  val heightPixels: Int

  /**
   * @return The width of the screen in pixels
   */

  val widthPixels: Int

}
